docker_tasks/build_stac/utils/stac.py

```diff
@@ -1,47 +1,38 @@
 import os
-from functools import singledispatch
-from pathlib import Path
 
 import pystac
 import rasterio
-from cmr import GranuleQuery
-from pystac.utils import str_to_datetime
 from rasterio.session import AWSSession
 from rio_stac import stac
 
 from . import events, regex, role
 
 
 def create_item(
+    item_id,
     properties,
     datetime,
-    cog_url,
     collection,
-    assets=None,
-    asset_name=None,
-    asset_roles=None,
-    asset_media_type=None,
+    assets,
 ) -> pystac.Item:
     """
     Function to create a stac item from a COG using rio_stac
     """
+    if "cog_default" in assets:
+        source = assets["cog_default"].href
+    else:
+        source = [asset.href for asset in assets][0]
```

Comment:
🤔

Reply:
dunno about this
maybe something like, if no asset is provided, create a `cog_default` asset (for backward compatibility)?

Reply:
At this point in the flow, all asset URLs should be associated with their asset names so I think the default inclusion of `cog_default` ought to happen earlier if we want to support that. This ought to, then, work nicely